src/fsharp/FSharp.Core/seq.fs

```diff
@@ -1381,24 +1381,40 @@ namespace Microsoft.FSharp.Collections
         [<CompiledName("Last")>]
         let last (source : seq<_>) =
             checkNonNull "source" source
-            use e = source.GetEnumerator()
-            if e.MoveNext() then
-                let mutable res = e.Current
-                while (e.MoveNext()) do res <- e.Current
-                res
-            else
-                invalidArg "source" LanguagePrimitives.ErrorStrings.InputSequenceEmptyString
+            match source with
+            | :? ('T[]) as a -> 
+                if a.Length=0 then invalidArg "source" LanguagePrimitives.ErrorStrings.InputSequenceEmptyString
```

Comment:
minor nitpick about coding style: `a.Length = 0`

Reply:
otherwise OK ?

Reply:
makes sense to me. yes

Reply:
> minor nitpick about coding style: `a.Length = 0`

fixed!